circuits/lib/query/valueCommitment.circom

```diff
@@ -0,0 +1,27 @@
+pragma circom 2.1.1;
+
+include "../../../node_modules/circomlib/circuits/mux1.circom";
+include "../../../node_modules/circomlib/circuits/comparators.circom";
+include "../../../node_modules/circomlib/circuits/poseidon.circom";
+
+/**
+Value commitment circuit allows to commit to a specific value and then
+reveal it later or use such a commitment in another circuits to prove that
+multiple circuits work with the same value without revealing it.
+*/
+
+template ValueCommitment() {
+    signal input value;
+    signal input commitNonce; // private random nonce to make the commitment unique and secure
+
+    signal output out;
+
+    signal isNonceZero <== IsZero()(commitNonce);
+
+    signal commit <== Poseidon(2)([value, commitNonce]);
```

Comment:
Looks like I can receive the same commitment if I have another field with the same value and provide the same commitment outside.  (I want to prove that I have the same date of birth in the different creds, but I can hack and prove that some another date that I have in other credential is equal to expected, but it is not the same meaning.

Reply:
It's not a hack. Maybe user wants to prove to verifier that on his day of birth there was a sun eclipse. It's ok.
We don't have a way to check meaning of fields and we shouldn't. User needs to check what verifier asks from him - does it make sense.